src/shared/openssl-util.c

```diff
@@ -109,6 +110,80 @@ int rsa_pkey_to_suitable_key_size(
         return 0;
 }
 
+static int pkey_generate_volume_key_rsa(
+                EVP_PKEY *pkey,
+                void **ret_decrypted_key,
+                size_t *ret_decrypted_key_size,
+                void **ret_savedata,
+                size_t *ret_savedata_size) {
```

Comment:
why did you name this `savedata`? why not stick to `encrypted_key`? I think that makes the logic a lot more obvious.